Magento2/Sniffs/Annotation/MethodArgumentsSniff.php

```diff
@@ -617,6 +622,96 @@ public function process(File $phpcsFile, $stackPtr)
         );
     }
 
+    /**
+     * Check method if it has defined return & parameter types, then it doesn't need DocBlock
+     *
+     * @param File $phpcsFile
+     * @param int $stackPtr
+     * @return bool
+     */
+    private function checkIfMethodNeedsDocBlock(File $phpcsFile, $stackPtr) : bool
+    {
+        if ($this->checkIfNamespaceContainsApi($phpcsFile)) {
+            // Note: API classes MUST have DocBlock because it uses them instead of reflection for types
+            return true;
+        }
```

Comment:
Please add a test to cover this case - where a docblock is required due to a class being an API.